runtime/src/snapshot_utils.rs

```diff
@@ -498,14 +512,7 @@ pub fn archive_snapshot_package(
         }
     }
 
-    // Write version file
-    {
-        let mut f = fs::File::create(&staging_version_file).map_err(|e| {
-            SnapshotError::IoWithSourceAndFile(e, "create version file", staging_version_file)
-        })?;
-        f.write_all(snapshot_package.snapshot_version.as_str().as_bytes())
-            .map_err(|e| SnapshotError::IoWithSource(e, "write version file"))?;
-    }
+    write_snapshot_version_file(staging_version_file, snapshot_package.snapshot_version)?;
```

Comment:
To be safe, I am not touching the existing working archiving/unchiving functions for now.

On this part, snapshot_version is passed in as a field in SnapshotInfo.  The other parameters are full_snapshot_archives_dir, incremental_snapshot_archives_dir.  But there is no info about the bank snapshot path.   To do that, I will need to do some plumbing work to pass the existing version file path here.   It is not trivial.

To address this duplication issue, I am think to do a much larger scale cleanup later.

I think we could separate these functions into two pipeline stages:
data structures in memory <-> snapshot in directory  <-> archives

Since the directory is a full representation of the snapshot state, we can build the archive directly from the directory, with no need of any data structures in memory, so SnapshotPackage, AccountsPackage could be gone.  The staging archiving directory will also be gone.  The new archiving function will be much simpler.

Reply:
fixed it.